docs/source/app_developers_guide/testing.rst

```diff
@@ -0,0 +1,145 @@
+*******************************
+Testing a Transaction Processor
+*******************************
+
+Say we've written a transaction processor -- how can it be tested? In this section, we'll see how to test the XO transaction processor described in :doc:`txn_family_tutorial` against a mock validator.
+
+Communication with the Validator
+================================
+
+First, let's look at what happens when the XO transaction 'create game000' is processed. Communication between the validator and a transaction processor (TP) takes place in six steps:
+
+1) The validator needs to process a transaction, so it sends a transaction process request to the TP containing the transaction information (in this case, the command 'create' and the game name 'game000'.
+
+2) The TP needs to know whether there already exists a game with the name 'game000', so it sends a 'get' request back to the validator.
+
+3) After querying state, the validator sends back the results to the TP (in this case, we'll assume that there is no game called 'game000').
+
+4) The TP now sends a message back to the validator requesting that it set 'game000' to an initialized game (with an empty board, etc).
+
+5) The validator sends the address of the created game back to the TP.
+
+6) The TP sends back an 'OK' response.
```

Comment:
The TP sends an 'OK' response to the validator, signaling that the transaction was processed successfully.